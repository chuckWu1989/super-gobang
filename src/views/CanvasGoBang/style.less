.canvas-gobang-style {
  width: 100%;
  position: relative;
}
