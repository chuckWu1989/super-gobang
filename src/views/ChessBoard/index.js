import React from 'react';

const ChessBoard = props => (
  <div>Test</div>
);

export default ChessBoard;
