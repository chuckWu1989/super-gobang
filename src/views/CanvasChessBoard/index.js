import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { drawBoard } from '../../controllers/CanvasChessBoard';
import './style.less';

class CanvasChessBoard extends Component {
  static propTypes = {
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }
  componentDidMount() {
    drawBoard(this.chessboard, this.props);
  }
  componentDidUpdate() {
    drawBoard(this.chessboard, this.props);
  }
  render() {
    const { width, height } = this.props;
    return (
      <canvas
        width={width}
        height={height}
        ref={(self) => { this.chessboard = self; }}
        className="canvas-chessboard-style"
      />
    );
  }
}

export default CanvasChessBoard;
