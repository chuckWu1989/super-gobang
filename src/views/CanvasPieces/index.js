import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './style.less';

class CanvasPieces extends Component {
  static defaultProps = {
    detectChessMovesChange: () => {},
    detectWidthChange: () => {},
  }
  static propTypes = {
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
    detectChessMovesChange: PropTypes.func,
    detectWidthChange: PropTypes.func,
  }
  componentWillReceiveProps(nextProps) {
    this.props.detectChessMovesChange(this.canvas, nextProps);
  }
  componentDidUpdate(prevProps) {
    this.props.detectWidthChange(this.canvas, prevProps);
  }
  render() {
    const { width, height } = this.props;
    return (
      <canvas
        target
        ref={(self) => { this.canvas = self; }}
        width={width}
        height={height}
        className="canvas-pieces-style"
      />
    );
  }
}

export default CanvasPieces;
