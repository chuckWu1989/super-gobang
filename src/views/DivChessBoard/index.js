import React from 'react';
import PropTypes from 'prop-types';
import './style.less';

const DivChessBoard = (props) => {
  const {
    width,
    height,
    grid,
    style,
    unitHeight,
  } = props;
  return (
    <div
      className="div-chessboard-style"
      style={{ width, height, ...style }}
    >
      {
        Array.from({ length: (grid + 1) }, (v, i) => (
          <div key={`${i}-board`} className="chessboard-row" style={{ height: unitHeight }}>
            {
              Array.from({ length: (grid + 1) }, (d, j) => (
                <div key={`${i}-${j}-board`} className="chessboard-col" />
              ))
            }
          </div>
        ))
      }
    </div>
  );
};
DivChessBoard.defaultProps = {
  grid: 15,
  style: {},
};
DivChessBoard.propTypes = {
  grid: PropTypes.number,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  unitHeight: PropTypes.number.isRequired,
  style: PropTypes.object,
};

export default DivChessBoard;
