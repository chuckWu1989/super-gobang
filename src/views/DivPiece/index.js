import React from 'react';
import PropTypes from 'prop-types';
import './style.less';

const DivPiece = props => (
  <div
    className="div-piece-style"
    style={{
      top: props.y,
      left: props.x,
      width: props.size,
      height: props.size,
      borderRadius: props.size / 2,
      backgroundColor: props.color,
    }}
  />
);
DivPiece.defaultProps = {
  size: 10,
  color: '#000000',
};
DivPiece.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
};

export default DivPiece;
