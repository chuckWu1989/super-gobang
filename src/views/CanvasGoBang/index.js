import React from 'react';
import CanvasChessBoard from '../../router/CanvasChessBoard';
import CanvasPieces from '../../router/CanvasPieces';
import geocoding from '../../router/geocoding';
import './style.less';

export const CanvasGoBang = props => (
  <div className="canvas-gobang-style">
    <CanvasChessBoard {...props} />
    <CanvasPieces {...props} />
  </div>
);

export default geocoding(CanvasGoBang);
