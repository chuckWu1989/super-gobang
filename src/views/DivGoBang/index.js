import React from 'react';
import DivChessBoard from '../../views/DivChessBoard';
import DivPieces from '../../router/DivPieces';
import geocoding from '../../router/geocoding';
import './style.less';

export const DivGoBang = props => (
  <div className="div-gobang-style">
    <DivChessBoard {...props} />
    <DivPieces {...props} />
  </div>
);

export default geocoding(DivGoBang);
