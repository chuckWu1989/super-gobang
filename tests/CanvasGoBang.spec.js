import React from 'react';
import toJson from 'enzyme-to-json';
import { CanvasGoBang } from '../src/views/CanvasGoBang';

const { shallow } = global;

describe('CanvasGoBang', () => {
  describe('view', () => {
    it('should be rendered properly', () => {
      const tree = toJson(shallow(<CanvasGoBang />));
      expect(tree).toMatchSnapshot();
    });
  });
});
