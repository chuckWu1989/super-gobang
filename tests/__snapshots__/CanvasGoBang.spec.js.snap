// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasGoBang view should be rendered properly 1`] = `
<div
  className="canvas-gobang-style"
>
  <Component />
  <Component />
</div>
`;
